app/models/goal.py

```diff
@@ -2,4 +2,16 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    tasks = db.relationship("Task", backref='goal', lazy=True)
+
+    def to_dict(self):
+        return {
+            "id": self.goal_id,
+            "title": self.title
+        }
+
+    @classmethod
+    def from_dict(cls, goal_data):
+        return cls(title=goal_data["title"])
```

Comment:
Remember to always put a newline at the end of the file -- this is good style practice in Python and a requirement in some other languages.